### PR TITLE
Remove import redundancy; Fix project setting UI

### PR DESCRIPTION
**Pull request by wisechengyi**
## Problem

Once the project is created, under Settings -> Build Tools -> Pants, Linked Pants project settings are not working. Also 3 pants calls were made in order to determine the targets under project path, which is 2 more than necessary.

E.g.  Imported targets are now marked incorrectly, and any attempt to change the setting will be ignored.
screen shot 2017-01-18 at 9 56 23 am

## Solution
screen shot 2017-01-18 at 10 31 26 am

Other change:
* Move `Use IDEA project JDK for Pants compilation` setting into Project setting. Earlier it is in a separate form, which causes confusion.
* Remove unused "load depedees" option.

### CODE REVIEW

File: src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java [hunk 1]
```diff
@@ -35,103 +36,121 @@
...
-    final String externalProjectPath = getInitialSettings().getExternalProjectPath();
-    if (!StringUtil.isEmpty(externalProjectPath)) {
-      onProjectPathChanged(externalProjectPath);
-    }
   }
 
   public void onProjectPathChanged(@NotNull final String projectPath) {
-    myTargetSpecs.clear();
+    if (lastPath.equals(projectPath)) {
+      lastPath = projectPath;
```
Comment (baroquebobcat) [inline src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java]: Why assign here if they are equal? This could be reduced to being just a guard clause.

File: src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java [hunk 2]
```diff
@@ -35,103 +36,121 @@
...
+    final JBLabel targetsLabel = new JBLabel(PantsBundle.message("pants.settings.text.targets"));
+    content.add(targetsLabel, lineConstraints);
 
-    myTargetSpecs = new CheckBoxList<String>();
-    content.add(ScrollPaneFactory.createScrollPane(myTargetSpecs), ExternalSystemUiUtil.getFillLineConstraints(0));
+    JScrollPane pane = ScrollPaneFactory.createScrollPane(myTargetSpecsBox);
+    content.add(pane, lineConstraints);
+  }
+
+  // It is silly `CheckBoxList` does not provide an iterator.
```
Comment (baroquebobcat) [inline src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java]: You could add a static external iterator method that accepts a java.util.function.Consumer. Then you could simplify this and the other loop in applyExtraSettings.

File: tests/com/twitter/intellij/pants/integration/PantsCompileOptionsExecutorTest.java [hunk 3]
```diff
@@ -18,7 +18,7 @@ public void testProjectName() throws Throwable {
 
     PantsExecutionSettings settings = new PantsExecutionSettings(
       Collections.singletonList(deepDir),
-      false, // with dependees. does not matter here
+      // with dependees. does not matter here
```
Comment (baroquebobcat) [inline tests/com/twitter/intellij/pants/integration/PantsCompileOptionsExecutorTest.java]: xx

File: tests/com/twitter/intellij/pants/settings/PantsProjectSettingsTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+
+import com.intellij.ui.CheckBoxList;
+import com.intellij.util.containers.ContainerUtil;
+import com.twitter.intellij.pants.testFramework.OSSPantsIntegrationTest;
+
+import java.io.File;
+
+public class PantsProjectSettingsTest extends OSSPantsIntegrationTest {
+  public void testDirectory() {
```
Comment (baroquebobcat) [inline tests/com/twitter/intellij/pants/settings/PantsProjectSettingsTest.java]: This first test is a little hard to follow.

How would you feel about breaking out some utility methods for the common components between it and testBuildFile. I'd also like to see some tests for the other states the control can be in, ie missing a file or non-pants project file.

File: src/com/twitter/intellij/pants/components/impl/PantsProjectComponentImpl.java [hunk 5]
```diff
@@ -113,7 +113,7 @@ private void convertToPantsProject() {
            */
           final List<String> targetSpecs = PantsUtil.gson.fromJson(serializedTargets, PantsUtil.TYPE_LIST_STRING);
           final PantsProjectSettings pantsProjectSettings =
-            new PantsProjectSettings(targetSpecs, projectPath, false, true, false);
+            new PantsProjectSettings(targetSpecs, projectPath, true, false, false);
```
Comment (wisechengyi) [inline src/com/twitter/intellij/pants/components/impl/PantsProjectComponentImpl.java]: corrected.

File: src/com/twitter/intellij/pants/settings/PantsExecutionSettings.java [hunk 6]
```diff
@@ -12,44 +12,38 @@
...
 
 public class PantsExecutionSettings extends ExternalSystemExecutionSettings implements PantsExecutionOptions {
-  private final boolean myWithDependees;
   private final boolean myLibsWithSourcesAndDocs;
   private final boolean myUseIdeaProjectJdk;
   private final boolean myEnableIncrementalImport;
   private final List<String> myTargetSpecs;
 
   private static final List<String> DEFAULT_TARGET_SPECS = Collections.emptyList();
-  private static final boolean DEFAULT_WITH_DEPENDEES = false;
```
Comment (wisechengyi) [inline src/com/twitter/intellij/pants/settings/PantsExecutionSettings.java]: corrected.

**Comment by wisechengyi** [comment]
Added `Remove unused "load depedees" option.` under description. The function was implemented, but the GUI never provided a way to configure it.

File: tests/com/twitter/intellij/pants/settings/PantsProjectSettingsTest.java (hunk 4 above)
Comment (wisechengyi) [inline tests/com/twitter/intellij/pants/settings/PantsProjectSettingsTest.java]: Added some negative test cases, separate and doc'ed util methods.

File: tests/com/twitter/intellij/pants/integration/PantsCompileOptionsExecutorTest.java (hunk 3 above)
Comment (wisechengyi) [inline tests/com/twitter/intellij/pants/integration/PantsCompileOptionsExecutorTest.java]: corrected.

File: src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java (hunk 2 above)
Comment (wisechengyi) [inline src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java]: Good call! Fixed

File: src/com/twitter/intellij/pants/settings/PantsExecutionSettings.java (hunk 6 above)
Comment (wisechengyi) [inline src/com/twitter/intellij/pants/settings/PantsExecutionSettings.java]: `DEFAULT_WITH_DEPENDEES` itself is removed because it is never used, but the readability is applied elsewhere.

**Comment by wisechengyi** [comment]
Thanks gents!
